File: pos_desktop/lib/models/product.dart
class Product {
  final String id;
  final String sku;
  final String name;
  final String description;
  final double price;
  final String imageUrl;
  final String category;
  final int stockCount;
  final String? supplierId;
  final String? size;
  final String? sizeNumeric;
  final double costPrice;

  /// Alias for costPrice — matches the SQLite backend column `buying_price`.
  double get buyingPrice => costPrice;

  Product({
    required this.id,
    required this.sku,
    required this.name,
    required this.description,
    required this.price,
    required this.imageUrl,
    required this.category,
    this.stockCount = 0,
    this.supplierId,
    this.size,
    this.sizeNumeric,
    this.costPrice = 0.0,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['id']?.toString() ?? json['_id']?.toString() ?? '',
      sku: json['sku']?.toString() ?? '',
      name: json['name']?.toString() ?? '',
      description: json['description']?.toString() ?? '',
      price: double.tryParse(json['price']?.toString() ?? '0') ?? 0.0,
      imageUrl: json['image_url']?.toString() ?? json['imageUrl']?.toString() ?? '',
      category: json['category']?.toString() ?? 'Uncategorized',
      stockCount: int.tryParse(json['stock_quantity']?.toString() ?? json['stock_count']?.toString() ?? json['stockCount']?.toString() ?? '0') ?? 0,
      supplierId: json['supplier_id']?.toString(),
      size: (json['size'] == null || json['size'] == 'null') ? null : json['size'].toString(),
      sizeNumeric: (json['size_numeric'] == null || json['size_numeric'] == 'null') ? null : json['size_numeric'].toString(),
      // Read buying_price (SQLite) OR cost_price (legacy) — whichever is present
      costPrice: double.tryParse(json['buying_price']?.toString() ?? json['cost_price']?.toString() ?? '0') ?? 0.0,
    );
  }
}

File: models/Product.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');
const Supplier = require('./Supplier');

const Product = sequelize.define('Product', {
    id: {
        type: DataTypes.INTEGER,
        primaryKey: true,
        autoIncrement: true
    },
    sku: {
        type: DataTypes.STRING(100),
        unique: true,
        allowNull: false
    },
    name: {
        type: DataTypes.STRING(255),
        allowNull: false
    },
    description: {
        type: DataTypes.TEXT
    },
    category: {
        type: DataTypes.STRING(255),
        defaultValue: 'Uncategorized'
    },
    price: {
        type: DataTypes.DECIMAL(10, 2),
        allowNull: false,
        defaultValue: 0.00
    },
    cost_price: {
        type: DataTypes.DECIMAL(10, 2),
        defaultValue: 0.00
    },
    stock_quantity: {
        type: DataTypes.INTEGER,
        defaultValue: 0
    },
    low_stock_threshold: {
        type: DataTypes.INTEGER,
        defaultValue: 10
    },
    woocommerce_product_id: {
        type: DataTypes.INTEGER,
        unique: true
    },
    regular_price: {
        type: DataTypes.DECIMAL(10, 2),
        allowNull: true
    },
    sale_price: {
        type: DataTypes.DECIMAL(10, 2),
        allowNull: true
    },
    stock_status: {
        type: DataTypes.STRING(50),
        defaultValue: 'instock'
    },
    image_url: {
        type: DataTypes.STRING(2048), // URL length
        allowNull: true
    },
    size: {
        type: DataTypes.STRING(10),
        allowNull: true
    },
    size_numeric: {
        type: DataTypes.STRING(20),
        allowNull: true
    },
    supplier_id: {
        type: DataTypes.INTEGER,
        references: {
            model: 'supply_chain_suppliers',
            key: 'id'
        }
    }
}, {
    tableName: 'inventory_products',
    timestamps: true,
    paranoid: true
});

// Setup relationships
Product.belongsTo(Supplier, { foreignKey: 'supplier_id' });
Supplier.hasMany(Product, { foreignKey: 'supplier_id' });

module.exports = Product;
